raster/src/main/scala/geotrellis/raster/reproject/RasterRegionReproject.scala

```diff
@@ -38,8 +41,30 @@ trait RasterRegionReproject[T <: CellGrid] extends Serializable {
     */
   def regionReproject(raster: Raster[T], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Raster[T]
 
-  def mutableRegionReproject(target: T, raster: Raster[T], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Unit
+  /** Reproject raster to a region that may partially intersect target raster extent.
+    *
+    * Back-projects only the cells overlapping the destination region before sampling their value from the source raster.
+    * This can be used to avoid assigning destination cell values where there is insufficient data for resample kernel.
+    *
+    * This is a <b>mutable</b> version of the region reproject. Target raster will be converted to mutable form and updated.
+    * In cases where target raster is subclass [[MutableArrayTile]] its values will be mutated.
+    * Its the responsability of the caller to use this function only when possible update in place is safe.\
+    * Correct use of this function allows to avoid intermidate memory allocation when accumulating results of multiple reproject calls.
+    *
+    * @param raster source raster to be sampled for reprojection.
+    * @param src source raster CRS
+    * @param dest target raster CRS
+    * @param target target raster that will be updated with pixels from source raster
+    * @param region polygon boundry of source raster in target CRS
+    * @param resampleMethod cell value resample method
+    */
+  def regionReprojectMutable(raster: Raster[T], src: CRS, dest: CRS, target: Raster[T], region: Polygon, resampleMethod: ResampleMethod): Raster[T]
```

Comment:
Strange convention but needed to find a name that would not break binary compatibility.